File: orchestrator-java-reactive/src/main/java/com/github/mzr/coder/orchestrator/java/reactive/service/OrchestrationService.java
package com.github.mzr.coder.orchestrator.java.reactive.service;


import com.github.mzr.coder.model.orhestrator.OrchestratorRequest;
import com.github.mzr.coder.model.orhestrator.OrchestratorResponse;
import reactor.core.publisher.Mono;

public interface OrchestrationService {
     Mono<OrchestratorResponse> getDecision(OrchestratorRequest request);
}

File: orchestrator-kotlin-coroutine/src/main/java/com/github/mzr/coder/orchestrator/kotlin/reactive/service/OrchestrationService.kt
package com.github.mzr.coder.orchestrator.kotlin.reactive.service

import com.github.mzr.coder.model.orhestrator.OrchestratorRequest
import com.github.mzr.coder.model.orhestrator.OrchestratorResponse
import reactor.core.publisher.Mono

interface OrchestrationService {
    suspend fun getDecision(request: OrchestratorRequest): OrchestratorResponse
}